### PR TITLE
pubsub/test: check for NotFound error code, and fail fast in conformance tests

### LINKED ISSUE #1791: pubsub: NotFound errors are not being returned correctly
The conformance tests check that an error is returned, but not that its code is `NotFound`. Several drivers are getting this wrong.

Update the tests and fix the drivers.

### PR DESCRIPTION
**Pull request by vangent**
* Check for `NotFound` error code in a couple of tests, and fix `mempubsub` to return it correctly.
* Update a couple of tests that `Receive` messages to time out faster if it isn't there (15s vs default 10m).

Fixes #1791.

### CODE REVIEW

**Comment by vangent** [comment]
Doh, it looks like a few other providers are not returning `NotFound` correctly... fixing.

**Comment by vangent** [comment]
PTAL

File: pubsub/awssnssqs/awssnssqs_test.go [hunk 1]
```diff
@@ -33,33 +33,10 @@ import (
 	"gocloud.dev/pubsub/drivertest"
 )
 
-const region = "us-east-2"
-
-func TestOpenTopic(t *testing.T) {
```
Comment (eliben) [inline pubsub/awssnssqs/awssnssqs_test.go]: So this went into drivertest?

Comment (vangent) [reply pubsub/awssnssqs/awssnssqs_test.go]: Yes, these two tests are already covered by conformance tests.

**Pull request merged**